File: CS/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using DevExpress.Utils;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Views.Card;
using DevExpress.XtraGrid.Views.Card.ViewInfo;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid;
using DevExpress.XtraEditors.Repository;

namespace HotTrack
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
		/// <summary>
		/// Required designer variable.
		/// </summary>
        private DevExpress.XtraGrid.GridControl gridControl1;
        private DevExpress.XtraGrid.Views.Card.CardView cardView1;
        private DevExpress.XtraEditors.Repository.RepositoryItemPictureEdit repositoryItemPictureEdit1;
        private nwindDataSet nwindDataSet;
        private BindingSource categoriesBindingSource;
        private HotTrack.nwindDataSetTableAdapters.CategoriesTableAdapter categoriesTableAdapter;
        private DevExpress.XtraGrid.Columns.GridColumn colCategoryID;
        private DevExpress.XtraGrid.Columns.GridColumn colCategoryName;
        private DevExpress.XtraGrid.Columns.GridColumn colDescription;
        private DevExpress.XtraGrid.Columns.GridColumn colPicture;
        private IContainer components;

		public Form1()
		{
			InitializeComponent();
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent() {
            this.components = new System.ComponentModel.Container();
            this.gridControl1 = new DevExpress.XtraGrid.GridControl();
            this.categoriesBindingSource = new System.Windows.Forms.BindingSource(this.components);
            this.nwindDataSet = new HotTrack.nwindDataSet();
            this.cardView1 = new DevExpress.XtraGrid.Views.Card.CardView();
            this.colCategoryID = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colCategoryName = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colDescription = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colPicture = new DevExpress.XtraGrid.Columns.GridColumn();
            this.repositoryItemPictureEdit1 = new DevExpress.XtraEditors.Repository.RepositoryItemPictureEdit();
            this.categoriesTableAdapter = new HotTrack.nwindDataSetTableAdapters.CategoriesTableAdapter();
            ((System.ComponentModel.ISupportInitialize)(this.gridControl1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.categoriesBindingSource)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.nwindDataSet)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.cardView1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.repositoryItemPictureEdit1)).BeginInit();
            this.SuspendLayout();
            // 
            // gridControl1
            // 
            this.gridControl1.DataSource = this.categoriesBindingSource;
            this.gridControl1.Dock = System.Windows.Forms.DockStyle.Fill;
            this.gridControl1.EmbeddedNavigator.Name = "";
            this.gridControl1.Location = new System.Drawing.Point(0, 0);
            this.gridControl1.MainView = this.cardView1;
            this.gridControl1.Name = "gridControl1";
            this.gridControl1.RepositoryItems.AddRange(new DevExpress.XtraEditors.Repository.RepositoryItem[] {
            this.repositoryItemPictureEdit1});
            this.gridControl1.Size = new System.Drawing.Size(627, 434);
            this.gridControl1.TabIndex = 0;
            this.gridControl1.ViewCollection.AddRange(new DevExpress.XtraGrid.Views.Base.BaseView[] {
            this.cardView1});
            this.gridControl1.MouseLeave += new System.EventHandler(this.gridControl1_MouseLeave);
            // 
            // categoriesBindingSource
            // 
            this.categoriesBindingSource.DataMember = "Categories";
            this.categoriesBindingSource.DataSource = this.nwindDataSet;
            // 
            // nwindDataSet
            // 
            this.nwindDataSet.DataSetName = "nwindDataSet";
            this.nwindDataSet.SchemaSerializationMode = System.Data.SchemaSerializationMode.IncludeSchema;
            // 
            // cardView1
            // 
            this.cardView1.Columns.AddRange(new DevExpress.XtraGrid.Columns.GridColumn[] {
            this.colCategoryID,
            this.colCategoryName,
            this.colDescription,
            this.colPicture});
            this.cardView1.FocusedCardTopFieldIndex = 0;
            this.cardView1.GridControl = this.gridControl1;
            this.cardView1.Name = "cardView1";
            this.cardView1.OptionsBehavior.FieldAutoHeight = true;
            this.cardView1.VertScrollVisibility = DevExpress.XtraGrid.Views.Base.ScrollVisibility.Auto;
            this.cardView1.CustomDrawCardFieldValue += new DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventHandler(this.cardView1_CustomDrawCardFieldValue);
            this.cardView1.MouseMove += new System.Windows.Forms.MouseEventHandler(this.gridView1_MouseMove);
            // 
            // colCategoryID
            // 
            this.colCategoryID.Caption = "CategoryID";
            this.colCategoryID.FieldName = "CategoryID";
            this.colCategoryID.Name = "colCategoryID";
            this.colCategoryID.Visible = true;
            this.colCategoryID.VisibleIndex = 0;
            // 
            // colCategoryName
            // 
            this.colCategoryName.Caption = "CategoryName";
            this.colCategoryName.FieldName = "CategoryName";
            this.colCategoryName.Name = "colCategoryName";
            this.colCategoryName.Visible = true;
            this.colCategoryName.VisibleIndex = 1;
            // 
            // colDescription
            // 
            this.colDescription.Caption = "Description";
            this.colDescription.FieldName = "Description";
            this.colDescription.Name = "colDescription";
            this.colDescription.Visible = true;
            this.colDescription.VisibleIndex = 2;
            // 
            // colPicture
            // 
            this.colPicture.Caption = "Picture";
            this.colPicture.ColumnEdit = this.repositoryItemPictureEdit1;
            this.colPicture.FieldName = "Picture";
            this.colPicture.Name = "colPicture";
            this.colPicture.Visible = true;
            this.colPicture.VisibleIndex = 3;
            // 
            // repositoryItemPictureEdit1
            // 
            this.repositoryItemPictureEdit1.Name = "repositoryItemPictureEdit1";
            // 
            // categoriesTableAdapter
            // 
            this.categoriesTableAdapter.ClearBeforeFill = true;
            // 
            // Form1
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(627, 434);
            this.Controls.Add(this.gridControl1);
            this.Name = "Form1";
            this.Text = "Form1";
            this.Load += new System.EventHandler(this.Form1_Load);
            ((System.ComponentModel.ISupportInitialize)(this.gridControl1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.categoriesBindingSource)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.nwindDataSet)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.cardView1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.repositoryItemPictureEdit1)).EndInit();
            this.ResumeLayout(false);

		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		private void Form1_Load(object sender, System.EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Categories' table. You can move, or remove it, as needed.
            this.categoriesTableAdapter.Fill(this.nwindDataSet.Categories);
		}

        private GridCell hotTrackCell = null;
        private GridCell HotTrackCell {
            get {
                return hotTrackCell;
            }
            set {
                int prevHotTrackRow = hotTrackCell == null ? GridControl.InvalidRowHandle : hotTrackCell.RowHandle;
                if(value == null) return;
                if(value == null || value.RowHandle == GridControl.InvalidRowHandle || value.Column == null) { 
                    this.hotTrackCell = null;
                }
                else {
                    if( hotTrackCell == null || (value.Column != hotTrackCell.Column ||
                        value.RowHandle != hotTrackCell.RowHandle)) {
                        hotTrackCell = value;
                        cardView1.RefreshRow(prevHotTrackRow);
                        cardView1.RefreshRow(hotTrackCell.RowHandle);
                    }
                }
            }
        }

		private void gridView1_MouseMove(object sender, System.Windows.Forms.MouseEventArgs e) {
            CardView view = sender as CardView;
			CardHitInfo info = view.CalcHitInfo(new Point(e.X, e.Y));
            GridCell c = new GridCell(info.RowHandle, info.Column);
            if(info.HitTest == CardHitTest.FieldValue){
                HotTrackCell = new GridCell(info.RowHandle, info.Column);
                if(info.Column.RealColumnEdit is RepositoryItemPictureEdit)
                    gridControl1.BeginInvoke(new MethodInvoker(delegate() { gridControl1.Cursor = Cursors.Hand; }));
                else
                    gridControl1.Cursor = Cursors.Default;
            }
            else {
                HotTrackCell = null;
            }
		}
		private void gridControl1_MouseLeave(object sender, System.EventArgs e) {
            HotTrackCell = null;
		}
        
        private void cardView1_CustomDrawCardFieldValue(object sender, RowCellCustomDrawEventArgs e) {
            if(hotTrackCell == null) return;
            if(e.RowHandle == hotTrackCell.RowHandle && e.Column == hotTrackCell.Column &&
                e.Column.RealColumnEdit is RepositoryItemPictureEdit){
                e.Appearance.BackColor = Color.DarkBlue;
            }
        }
	}
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports DevExpress.Utils
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Views.Card
Imports DevExpress.XtraGrid.Views.Card.ViewInfo
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid
Imports DevExpress.XtraEditors.Repository

''' <summary>
''' Summary description for Form1.
''' </summary>
Public Class Form1
    Inherits System.Windows.Forms.Form
    ''' <summary>
    ''' Required designer variable.
    ''' </summary>
    Private WithEvents gridControl1 As DevExpress.XtraGrid.GridControl
    Private WithEvents cardView1 As DevExpress.XtraGrid.Views.Card.CardView
    Private repositoryItemPictureEdit1 As DevExpress.XtraEditors.Repository.RepositoryItemPictureEdit
    Private nwindDataSet As nwindDataSet
    Private categoriesBindingSource As BindingSource
    Private categoriesTableAdapter As nwindDataSetTableAdapters.CategoriesTableAdapter
    Private colCategoryID As DevExpress.XtraGrid.Columns.GridColumn
    Private colCategoryName As DevExpress.XtraGrid.Columns.GridColumn
    Private colDescription As DevExpress.XtraGrid.Columns.GridColumn
    Private colPicture As DevExpress.XtraGrid.Columns.GridColumn
    Private components As IContainer

    Public Sub New()
        InitializeComponent()
    End Sub

    ''' <summary>
    ''' Clean up any resources being used.
    ''' </summary>
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not components Is Nothing Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

#Region "Windows Form Designer generated code"
    ''' <summary>
    ''' Required method for Designer support - do not modify
    ''' the contents of this method with the code editor.
    ''' </summary>
    Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container()
        Me.gridControl1 = New DevExpress.XtraGrid.GridControl()
        Me.categoriesBindingSource = New System.Windows.Forms.BindingSource(Me.components)
        Me.nwindDataSet = New nwindDataSet()
        Me.cardView1 = New DevExpress.XtraGrid.Views.Card.CardView()
        Me.colCategoryID = New DevExpress.XtraGrid.Columns.GridColumn()
        Me.colCategoryName = New DevExpress.XtraGrid.Columns.GridColumn()
        Me.colDescription = New DevExpress.XtraGrid.Columns.GridColumn()
        Me.colPicture = New DevExpress.XtraGrid.Columns.GridColumn()
        Me.repositoryItemPictureEdit1 = New DevExpress.XtraEditors.Repository.RepositoryItemPictureEdit()
        Me.categoriesTableAdapter = New nwindDataSetTableAdapters.CategoriesTableAdapter()
        CType(Me.gridControl1, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.categoriesBindingSource, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.nwindDataSet, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.cardView1, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.repositoryItemPictureEdit1, System.ComponentModel.ISupportInitialize).BeginInit()
        Me.SuspendLayout()
        ' 
        ' gridControl1
        ' 
        Me.gridControl1.DataSource = Me.categoriesBindingSource
        Me.gridControl1.Dock = System.Windows.Forms.DockStyle.Fill
        Me.gridControl1.EmbeddedNavigator.Name = ""
        Me.gridControl1.Location = New System.Drawing.Point(0, 0)
        Me.gridControl1.MainView = Me.cardView1
        Me.gridControl1.Name = "gridControl1"
        Me.gridControl1.RepositoryItems.AddRange(New DevExpress.XtraEditors.Repository.RepositoryItem() {Me.repositoryItemPictureEdit1})
        Me.gridControl1.Size = New System.Drawing.Size(627, 434)
        Me.gridControl1.TabIndex = 0
        Me.gridControl1.ViewCollection.AddRange(New DevExpress.XtraGrid.Views.Base.BaseView() {Me.cardView1})
        '			Me.gridControl1.MouseLeave += New System.EventHandler(Me.gridControl1_MouseLeave);
        ' 
        ' categoriesBindingSource
        ' 
        Me.categoriesBindingSource.DataMember = "Categories"
        Me.categoriesBindingSource.DataSource = Me.nwindDataSet
        ' 
        ' nwindDataSet
        ' 
        Me.nwindDataSet.DataSetName = "nwindDataSet"
        Me.nwindDataSet.SchemaSerializationMode = System.Data.SchemaSerializationMode.IncludeSchema
        ' 
        ' cardView1
        ' 
        Me.cardView1.Columns.AddRange(New DevExpress.XtraGrid.Columns.GridColumn() {Me.colCategoryID, Me.colCategoryName, Me.colDescription, Me.colPicture})
        Me.cardView1.FocusedCardTopFieldIndex = 0
        Me.cardView1.GridControl = Me.gridControl1
        Me.cardView1.Name = "cardView1"
        Me.cardView1.OptionsBehavior.FieldAutoHeight = True
        Me.cardView1.VertScrollVisibility = DevExpress.XtraGrid.Views.Base.ScrollVisibility.Auto
        '			Me.cardView1.CustomDrawCardFieldValue += New DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventHandler(Me.cardView1_CustomDrawCardFieldValue);
        '			Me.cardView1.MouseMove += New System.Windows.Forms.MouseEventHandler(Me.gridView1_MouseMove);
        ' 
        ' colCategoryID
        ' 
        Me.colCategoryID.Caption = "CategoryID"
        Me.colCategoryID.FieldName = "CategoryID"
        Me.colCategoryID.Name = "colCategoryID"
        Me.colCategoryID.Visible = True
        Me.colCategoryID.VisibleIndex = 0
        ' 
        ' colCategoryName
        ' 
        Me.colCategoryName.Caption = "CategoryName"
        Me.colCategoryName.FieldName = "CategoryName"
        Me.colCategoryName.Name = "colCategoryName"
        Me.colCategoryName.Visible = True
        Me.colCategoryName.VisibleIndex = 1
        ' 
        ' colDescription
        ' 
        Me.colDescription.Caption = "Description"
        Me.colDescription.FieldName = "Description"
        Me.colDescription.Name = "colDescription"
        Me.colDescription.Visible = True
        Me.colDescription.VisibleIndex = 2
        ' 
        ' colPicture
        ' 
        Me.colPicture.Caption = "Picture"
        Me.colPicture.ColumnEdit = Me.repositoryItemPictureEdit1
        Me.colPicture.FieldName = "Picture"
        Me.colPicture.Name = "colPicture"
        Me.colPicture.Visible = True
        Me.colPicture.VisibleIndex = 3
        ' 
        ' repositoryItemPictureEdit1
        ' 
        Me.repositoryItemPictureEdit1.Name = "repositoryItemPictureEdit1"
        ' 
        ' categoriesTableAdapter
        ' 
        Me.categoriesTableAdapter.ClearBeforeFill = True
        ' 
        ' Form1
        ' 
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(627, 434)
        Me.Controls.Add(Me.gridControl1)
        Me.Name = "Form1"
        Me.Text = "Form1"
        '			Me.Load += New System.EventHandler(Me.Form1_Load);
        CType(Me.gridControl1, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.categoriesBindingSource, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.nwindDataSet, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.cardView1, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.repositoryItemPictureEdit1, System.ComponentModel.ISupportInitialize).EndInit()
        Me.ResumeLayout(False)

    End Sub
#End Region

    ''' <summary>
    ''' The main entry point for the application.
    ''' </summary>
    <STAThread()> _
    Shared Sub Main()
        Application.Run(New Form1())
    End Sub

    Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ' TODO: This line of code loads data into the 'nwindDataSet.Categories' table. You can move, or remove it, as needed.
        Me.categoriesTableAdapter.Fill(Me.nwindDataSet.Categories)
    End Sub

    Private hotTrackCell_Renamed As GridCell = Nothing
    Private Property HotTrackCell() As GridCell
        Get
            Return hotTrackCell_Renamed
        End Get
        Set(ByVal value As GridCell)
            Dim prevHotTrackRow As Integer
            If hotTrackCell_Renamed Is Nothing Then
                prevHotTrackRow = GridControl.InvalidRowHandle
            Else
                prevHotTrackRow = hotTrackCell_Renamed.RowHandle
            End If
            If value Is Nothing Then
                Return
            End If
            If value Is Nothing OrElse value.RowHandle = GridControl.InvalidRowHandle OrElse value.Column Is Nothing Then
                Me.hotTrackCell_Renamed = Nothing
            Else
                If hotTrackCell_Renamed Is Nothing OrElse (Not value.Column Is hotTrackCell_Renamed.Column OrElse value.RowHandle <> hotTrackCell_Renamed.RowHandle) Then
                    hotTrackCell_Renamed = value
                    cardView1.RefreshRow(prevHotTrackRow)
                    cardView1.RefreshRow(hotTrackCell_Renamed.RowHandle)
                End If
            End If
        End Set
    End Property

    Private Sub gridView1_MouseMove(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles cardView1.MouseMove
        Dim view As CardView = TryCast(sender, CardView)
        Dim info As CardHitInfo = view.CalcHitInfo(New Point(e.X, e.Y))
        Dim c As GridCell = New GridCell(info.RowHandle, info.Column)
        If info.HitTest = CardHitTest.FieldValue Then
            HotTrackCell = New GridCell(info.RowHandle, info.Column)
            If TypeOf info.Column.RealColumnEdit Is RepositoryItemPictureEdit Then
                gridControl1.BeginInvoke(New MethodInvoker(AddressOf AnonymousMethod1))
            Else
                gridControl1.Cursor = Cursors.Default
            End If
        Else
            HotTrackCell = Nothing
        End If
    End Sub
    Private Sub AnonymousMethod1()
        gridControl1.Cursor = Cursors.Hand
    End Sub
    Private Sub gridControl1_MouseLeave(ByVal sender As Object, ByVal e As System.EventArgs) Handles gridControl1.MouseLeave
        HotTrackCell = Nothing
    End Sub

    Private Sub cardView1_CustomDrawCardFieldValue(ByVal sender As Object, ByVal e As RowCellCustomDrawEventArgs) Handles cardView1.CustomDrawCardFieldValue
        If hotTrackCell_Renamed Is Nothing Then
            Return
        End If
        If e.RowHandle = hotTrackCell_Renamed.RowHandle AndAlso e.Column Is hotTrackCell_Renamed.Column AndAlso TypeOf e.Column.RealColumnEdit Is RepositoryItemPictureEdit Then
            e.Appearance.BackColor = Color.DarkBlue
        End If
    End Sub
End Class
